lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -71,6 +72,32 @@ def self.create_release(repository, version, release_notes, assets, prerelease)
           github_client().upload_asset(release[:url], file_path, content_type: 'application/octet-stream')
         end
       end
+
+      # Downloads a file from the given GitHub release
+      #
+      # @param [String] repository The repository name (including the organization)
+      # @param [String] relase The title of the GitHub release
+      # @param [String] file_path The path, inside the project folder, of the file to download
+      # @param [Striog] download_folder The folder which the file should be downloaded into
+      # @return [String] The path of the downloaded file, or nil if something went wrong
+      #
+      def self.download_file_from_release(repository:, release:, file_path:, download_folder:)
+        repository = repository.delete_prefix('/').chomp('/').concat('/')
+        file_path = file_path.delete_prefix('/').chomp('/').concat('/')
+        release = release.concat('/')
```

Comment:
Isn't `URI.join` supposed to add the `/` at the right places when needed for you? You're right to `delete_prefix`, since if any argument you pass to `URI.join` has a leading `/` it will take over as being considered the URI's path's root, but for the suffix there is no need to neither `chomp` any extra `/` nor `concat` one yourself.

```suggestion
        repository.delete_prefix!('/')
        file_path.delete_prefix!('/')
```

Reply:
Hmm... I thought so. But I added the `/` because `URI.join` didn't work as expected without them. 
I tried to follow your suggestion again in `07fb871`, but it failed. It looks like `URI.join` doesn't behave as expected in this case.

Reply:
This is because you used `delete_prefix` – and not `delete_prefix!` – in 07fb871, so not surprising it failed 😉 

`delete_prefix!` with a exclamation mark at the end – like in my suggestion above – means (and that's a convention in Ruby) that it will mutate the object this is called on, while the version without the exclamation mark returns a new modified value instead of mutating the original.

In other words, `x.delete_prefix!('/')` is a synonym for `x = x.delete_prefix('/')`.

Reply:
Isn't `x = x.delete_prefix('/')` the same of `repository = repository.delete_prefix('/')` and `file_path = file_path.delete_prefix('/')` which I used in 07fb871?

Reply:
🤦‍♂️🤦‍♂️ Sorry… I misread the diff of that commit the first time around 🤦 nvm then

Reply:
Yeah, no problem. I'm still surprised that `URI.join` doesn't work as expected tbh. 🤷‍♂️